geometric_kernels/spaces/homogeneous_spaces.py

```diff
@@ -0,0 +1,184 @@
+import abc
+
+import lab as B
+import numpy as np
+
+from geometric_kernels.spaces.base import DiscreteSpectrumSpace
+from geometric_kernels.spaces.eigenfunctions import (
+    Eigenfunctions,
+    EigenfunctionWithAdditionTheorem,
+)
+from geometric_kernels.spaces.lie_groups import LieGroupCharacter, MatrixLieGroup
+
+
+class CompactHomogeneousSpaceAddtitionTheorem(EigenfunctionWithAdditionTheorem):
+    def __init__(self, M, num_levels, H_samples):
+        self.M = M
+        self.dim = M.G.dim - M.H.dim
+        self.G_n = M.G.n
```

Comment:
Is this really necessary: `self.G_n` is the same length as "self.G.n"? Maybe clean up this a bit, there's so many fields in this class!